src/core/federation.ts

```diff
@@ -1597,63 +1691,36 @@ export class FederatedApp extends App {
 
     this.rtiClient.on(
       "message",
-      <T>(destPortAction: Action<T>, messageBuffer: Buffer) => {
-        // Schedule this federate port's action.
-        // This message is untimed, so schedule it immediately.
+      <T>(destPortID: number, messageBuffer: Buffer) => {
         Log.debug(this, () => {
           return "(Untimed) Message received from RTI.";
         });
         // TODO (axmmisaka): use a type-safe interface, like io.ts?
         // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
         const value: T = JSON.parse(messageBuffer.toString());
 
-        destPortAction
-          .asSchedulable(this._getKey(destPortAction))
-          .schedule(0, value);
+        for (const candidate of this.networkReceivers) {
```

Comment:
This is OK if we expect the number of receivers to always be small, but a map would be more efficient.

Reply:
A map is now used to store `networkReceiver`s in the commit [b8b6c73](https://github.com/lf-lang/reactor-ts/pull/186/commits/b8b6c73d48ea9c28b2ebcd0bd9fd1adaad84a891).